src/main/java/pingis/services/sandbox/SandboxService.java

```diff
@@ -46,7 +46,7 @@
   public boolean updateSubmissionResult(UUID submissionId, String testOutput, String stdout,
       String stderr, String validations, String vmLog, String status, int exitCode)
       throws IOException {
-    Submission submission = submissionRepository.findOne(submissionId);
+    Submission submission = submissionRepository.findById(submissionId).get();
 
     if (submission == null || submission.getStatus() != SubmissionStatus.PENDING) {
```

Comment:
You're right, in fact I missed this null check when mechanically replacing `findOne`s. If the optional is not present the `get` will cause a `NoSuchElementException`, so code like above doesn't work at all if the value isn't present. I added tests for two such scenarios as previously the tests passed even with this error. In the above I used `isPresent`, but in the other, like in `UserService`'s `findOne`, I just made it function like it used to by returning `null` if not present. Definitely not ideal, but it works.